### PR TITLE
Remove quarkus-bom-deployment deprecated in 1.6.0.Final

### PR DESCRIPTION
**Pull request by aloubyansky**
Removed deprecated quarkus-bom-deployment BOM.

### CODE REVIEW

File: extensions/pom.xml [hunk 1]
```diff
@@ -200,8 +200,8 @@
                     <artifactIdPrefix>quarkus-</artifactIdPrefix>
                     <namePrefix xml:space="preserve">Quarkus - </namePrefix>
                     <quarkusVersion>@{project.version}</quarkusVersion>
-                    <runtimeBomPath>../bom/runtime/pom.xml</runtimeBomPath>
-                    <deploymentBomPath>../bom/deployment/pom.xml</deploymentBomPath>
+                    <runtimeBomPath>../bom/1pp/pom.xml</runtimeBomPath>
```
Comment (gastaldi) [inline extensions/pom.xml:203]: Is this path correct? `1pp`?

Comment (gastaldi) [reply extensions/pom.xml:203]: Also shouldn't it be `../bom/application/pom.xml` for both?

Comment (aloubyansky) [reply extensions/pom.xml:203]: omg, thanks

Comment (aloubyansky) [reply extensions/pom.xml:203]: how did it not fail?.. Maven is weird.

Comment (gastaldi) [reply extensions/pom.xml:203]: This part seems to be commented out, maybe that's why?

Comment (aloubyansky) [reply extensions/pom.xml:203]: Ah, right, it's commented out. Yes, it should be the same path. I'll just update the plugin to remove the deploymentBomPath.

**Review by gastaldi — APPROVED** [review]
LGTM

**Review by gsmet — CHANGES_REQUESTED** [review]
I would like the Platform PR to be ready before merging this one: https://github.com/quarkusio/quarkus-platform/blob/master/bom/deployment/pom.xml#L29 .

Thanks!

**Comment by aloubyansky** [comment]
The PR removing `quarkus-bom-deployment` from the platform is https://github.com/quarkusio/quarkus-platform/pull/96

**Comment by aloubyansky** [comment]
@gsmet could you please review this one again when you get a chance? I moved the BOM from `bom/runtime` to `bom/application`. And now I have to rebase it a few times a day to avoid conflicts.

**Comment by gsmet** [comment]
To be honest, I’m a bit worried about doing that in the summer. 

What exactly will happen if we have extensions that don’t remove it? Will it fail big time or not?

**Comment by aloubyansky** [comment]
They won't build with 1.7.0.Final. To fix that the import of quarkus-bom-deployment will have to be removed.

**Review by gsmet — APPROVED** [review]

**Pull request merged**